manta-js/package/sdk.ts

```diff
@@ -0,0 +1,604 @@
+            const endTime = performance.now();
+            this.conditionalLog(`Initial sync finished in ${(endTime - startTime) / 1000} seconds`);
+            this.walletIsBusy = false;
+            this.initialSyncIsFinished = true;
+        } catch (e) {
+            this.walletIsBusy = false;
+            console.error("Initial sync failed.",e);
+        }
+    }
+
+    /// Pulls data from the ledger, synchronizing the currently connected wallet and
+    /// balance state. This method runs until all the ledger data has arrived at and
+    /// has been synchronized with the wallet.
+    async walletSync(): Promise<void> {
+        try {
+            if (!this.initialSyncIsFinished) {
+                console.error("Must call initalWalletSync before walletSync!");
+            }
+            await this.waitForWallet();
+            this.walletIsBusy = true;
+            this.conditionalLog('Beginning sync');
+            const networkType = this.wasm.Network.from_string(`"${this.network}"`);
+            const startTime = performance.now();
+            await this.wasmWallet.sync(networkType);
+            const endTime = performance.now();
+            this.conditionalLog(`Initial sync finished in ${(endTime - startTime) / 1000} seconds`);
+            this.walletIsBusy = false;
+        } catch (e) {
+            this.walletIsBusy = false;
+            console.error("Sync failed.",e);
+        }
+    }
+
+    /// Returns the private balance of the currently connected zkAddress for the currently
+    /// connected network.
+    async getPrivateBalance(assetId: BN): Promise<string> {
+        try {
+            await this.waitForWallet();
+            this.walletIsBusy = true;
+            const balance = await this.wasmWallet.balance(assetId.toString());
+            this.walletIsBusy = false;
+            return balance;
+        } catch (e) {
+            this.walletIsBusy = false;
+            console.error("Failed to fetch private balance.",e);
+        }
+    }
+
+    /// Returns the public balance associated with an account for a given `asset_id`.
+    /// If no address is provided, the balance will be returned for this.signer.
+    async getPublicBalance(assetId: BN, address:Address): Promise<any> {
+        try {
+            if (assetId.toString() === NATIVE_TOKEN_ASSET_ID) {
+                const nativeBalance: any = await this.api.query.system.account(address);
+                return nativeBalance.data.free.toHuman();
+            } else {
+                const assetBalance: any = await this.api.query.assets.account(assetId, address);
+                if (assetBalance.value.isEmpty) {
+                    return "0";
+                } else {
+                    return assetBalance.value.balance.toString();
+                }
+            }
+        } catch (e) {
+            console.log("Failed to fetch public balance.");
+            console.error(e);
+        }
+    }
```

Comment:
We should move this function out of this class/file and into q new file called `utils.ts` since it is not called from within `MantaPrivateWallet`, and not really private wallet functionality.